server/cluster_worker.go

```diff
@@ -233,24 +253,14 @@ func (c *raftCluster) maybeSplit(request *pdpb.RegionHeartbeatRequest, reqRegion
 		return nil, nil
 	}
 
-	var ops []clientv3.Op
-
-	regionValue, err := proto.Marshal(reqRegion)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	regionEncStartKey := encodeRegionSearchKey(region.GetStartKey())
+	regionEncStartKey := encodeRegionStartKey(region.GetStartKey())
 	regionEncEndKey := encodeRegionSearchKey(region.GetEndKey())
-	reqRegionEncStartKey := encodeRegionSearchKey(reqRegion.GetStartKey())
-	reqRegionEncEndKey := encodeRegionSearchKey(reqRegion.GetEndKey())
+
 	if reqRegionEncStartKey == regionEncStartKey &&
 		reqRegionEncEndKey == regionEncEndKey {
 		// Seems there is something wrong? Do nothing.
```

Comment:
@siddontang In that case, the 'epoch.Version' will be unchanged and this func will return at L253 above.